src/main/java/com/google/devtools/build/lib/skyframe/BuildTopLevelAspectsDetailsFunction.java

```diff
@@ -88,13 +89,19 @@ public SkyValue compute(SkyKey skyKey, Environment env)
       // duplicate aspects by keeping the first occurrence of each aspect.
       topLevelAspectsClasses = ImmutableSet.copyOf(topLevelAspectsClasses).asList();
 
-      // Then return a list of indenpendent aspects to be applied on the top-level targets
-      ImmutableList<AspectDetails> aspectsDetailsList =
-          getIndependentTopLevelAspects(env, topLevelAspectsClasses);
-      if (aspectsDetailsList == null) {
-        return null; // some aspects are not loaded
+      try {
+        // Then return a list of indenpendent aspects to be applied on the top-level targets
+        ImmutableList<AspectDetails> aspectsDetailsList =
+            getIndependentTopLevelAspects(env, topLevelAspectsClasses);
+        if (aspectsDetailsList == null) {
+          return null; // some aspects are not loaded
+        }
+        return new BuildTopLevelAspectsDetailsValue(aspectsDetailsList);
+      } catch (AspectCreationException e) {
```

Comment:
Can we add this exception catch to `getIndependentTopLevelAspects` to be similar to `getTopLevelAspects`?

Reply:
Done, now they are more similar.